distributed/scheduler.py

```diff
@@ -5161,7 +5165,9 @@ async def restart(self, client=None, timeout=30):
 
         self.log_event([client, "all"], {"action": "restart", "client": client})
         start = time()
-        while time() < start + 10 and len(self.workers) < n_workers:
+        while time() < start + 10 and (
+            len(self.workers) < n_workers or initial_workers & self._get_worker_ids()
```

Comment:
See this comment: https://github.com/dask/distributed/pull/6637/files#diff-bbcf2e505bf2f9dd0dc25de4582115ee4ed4a6e80997affc7b22122912cc6591R5108-R5109

Workers without nannies will be closed and not come back. So we shouldn't wait for them. We should just be waiting for `nanny_workers`.

Reply:
The way I understood the intent of that comment and the code, we purposefully rely on the deployment system to bring the workers back up and we therefore want to wait for that to happen. Otherwise, in a non-nanny environment, `Scheduler.restart` would just be a Scheduler.shutdown` with extra steps. I might be wrong though...and we should definitely improve the way we handle the case where this does not happen.

Reply:
This is related to Florian's comment here: https://github.com/dask/distributed/pull/6637#discussion_r909503067